packages/store/src/secure-store/index.ts

```diff
@@ -0,0 +1,50 @@
+import * as Keytar from 'keytar';
+import { Logger } from '@sap-ux/logger';
+import { isAppStudio, errorString } from '../utils';
+import { DummyStore } from './dummy-store';
+import { KeytarStore } from './keytar-store';
+import { SecureStore } from './types';
+
+function getKeytar(log: Logger): typeof Keytar | undefined {
+    try {
+        return require('keytar');
+    } catch (err) {
+        log.warn(errorString(err));
+        log.warn(`Could not "require('keytar')". Trying VSCode's copy`);
+        let vscode;
+        try {
+            vscode = require('vscode');
+        } catch (e) {
+            log.warn(errorString(e));
+            log.warn('Could not get hold of vscode');
+            return undefined;
+        }
+        try {
+            return require(`${vscode?.env?.appRoot}/node_modules.asar/keytar`);
```

Comment:
Do we need to check version compatibility here?

Reply:
Not really. This comes from here: https://github.com/microsoft/vscode/issues/68738#issuecomment-470005069
This is why we try getting access to `keytar` 3 different ways and it's coded so defensively.

I can take another look at [vscode-pull-request-github](https://github.com/microsoft/vscode-pull-request-github) and see how they may have evolved loading keytar.